app/facades/games_facade.rb

```diff
@@ -0,0 +1,48 @@
+class GamesFacade
+	class << self
+
+    def repurpose(params)
+      if params[:extension]
+        game = FriendlyGame.find_by(extension: params[:extension])
+        if next_game = game.next_game
+          next_game
+        else
+          if game.status == ('drawn' || 3)
+            new_game = FriendlyGame.create(starting_fen: game.starting_fen, white_id: game.white_id, black_id: game.black_id)
+          else
+            new_game = new_game(game)
+          end
+          game.next_game_id = new_game.id
+          game.save
+          new_game
+        end
+      else
+        fen = Fen.new().to_starting_position
+        FriendlyGame.create(starting_fen: fen.fen)
+      end
+    end
+
+    def add_player?(ext, api)
+      game = FriendlyGame.find_by(extension: ext) rescue nil
+      id = User.find_by(api_key: api).id rescue nil
+      if id.nil? || game.nil? or game.black_id or game.white_id == id
+        true
+      else
+        game.white_id ? game.black_id = id : game.white_id = id
+        game.save
+      end
+    end
+
```

Comment:
Here is the add_player? method used in the app/channels/friendly_games_channel.rb. Currently this is not optimized, and there are a couple ways to go to save server time if this app were to scale:

- When a spectator watches a game, maybe we should catch that the game already has two players, and does not need to even call this method.
- Currently, only the first game between two players needs to add players. Every subsequent game will fill in automatically from the first one. However, eventually in random matchmaking this will be more relevant.